File: app/src/main/java/com/sagalogistics/backend/models/Item.java
package com.sagalogistics.backend.models;

import androidx.annotation.NonNull;

import java.io.Serializable;

public interface Item extends Serializable, Cloneable {
    String getKey();

    void setKey(String key);

    String getName();

    void setName(String name);

    Float getWeight();

    void setWeight(Float weight);

    @NonNull
    Object clone() throws CloneNotSupportedException;
}

File: app/src/main/java/com/sagalogistics/backend/api/models/Item.kt
package com.sagalogistics.backend.api.models

import java.io.Serializable

/**
 * An item
 */
interface Item : Serializable {
    /**
     * The key of the item in the database
     */
    var key: String?

    /**
     * The name of the item
     */
    var name: String

    /**
     * The weight of the item
     */
    var weight: Float

    /**
     * The upper variance percentage in weight of the item
     *
     * Must be between 0 and 100
     */
    var upperVariance: Float

    /**
     * The lower variance percentage in weight of the item
     *
     * Must be between 0 and 100
     */
    var lowerVariance: Float

    /**
     * The url of an image representing the item
     */
    var image: String?
}